EFCore/ASP.NetCore/Blazor.WebAssembly/Readme.md

```diff
@@ -0,0 +1,1114 @@
+      }
+      else {
+         string sqliteDBPath = Path.Join(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "WebAPIDemo.db");
+         options.UseSqlite($"Data Source={sqliteDBPath}");
+      }
+      // ...
+   }
+}
+```
+
+### Generate initial data
+
+1. Open the `WebAPI/DatabaseUpdate/Updater.cs` file and add the following code to the `UpdateDatabaseAfterUpdateSchema` method to create appropriate users (Editor and Viewer), assign roles/permissions, and create sample Post objects:
+
+   _DatabaseUpdate\Updater.cs_:
+   ```cs
+   var editorUser = ObjectSpace.FirstOrDefault<ApplicationUser>(user=>user.UserName=="Editor")??ObjectSpace.CreateObject<ApplicationUser>();
+   if (ObjectSpace.IsNewObject(editorUser)) {
+      //create Editor User/Role
+      editorUser.UserName="Editor";
+      editorUser.Email = "Editor@example.com";
+
+      var editorRole = ObjectSpace.CreateObject<PermissionPolicyRole>();
+      editorRole.Name = "EditorRole";
+      editorRole.AddTypePermission<Post>(SecurityOperations.CRUDAccess, SecurityPermissionState.Allow);
+      editorRole.AddTypePermission<ApplicationUser>(SecurityOperations.CRUDAccess, SecurityPermissionState.Allow);
+      editorUser.Roles.Add(editorRole);
+      editorUser.Roles.Add(defaultRole);
+      editorUser.Photo = ObjectSpace.CreateObject<MediaDataObject>();
+      editorUser.Photo.MediaData = GetResourceByName("Janete");
+
+      //create Viewer User/Role
+      var viewerUser = ObjectSpace.CreateObject<ApplicationUser>();
+      viewerUser.UserName = "Viewer";
+      viewerUser.Email = "Viewer@example.com";
+
+      viewerUser.Photo = ObjectSpace.CreateObject<MediaDataObject>();
+      viewerUser.Photo.MediaData = GetResourceByName("John");
+      var viewerRole = ObjectSpace.CreateObject<PermissionPolicyRole>();
+      viewerRole.Name = "ViewerRole";
+      viewerRole.AddTypePermission<Post>(SecurityOperations.Read, SecurityPermissionState.Allow);
+      viewerRole.AddTypePermission<ApplicationUser>(SecurityOperations.Read, SecurityPermissionState.Allow);
+      viewerUser.Roles.Add(viewerRole);
+      viewerUser.Roles.Add(defaultRole);
+
+      //commit
+      ObjectSpace.CommitChanges();
+
+      //assign authentication type
+      foreach (var user in new[] { editorUser, viewerUser }.Cast<ISecurityUserWithLoginInfo>()) {
+         user.CreateUserLoginInfo(SecurityDefaults.PasswordAuthentication,
+            ObjectSpace.GetKeyValueAsString(user));
+      }
+
+      //sample posts
+      var post = ObjectSpace.CreateObject<Post>();
+      post.Title = "Hello World";
+      post.Content = "This is a FREE API for everybody";
+      post.Author=editorUser;
+      post = ObjectSpace.CreateObject<Post>();
+      post.Title = "Hello Blazor WebAssembly";
+      post.Content = "Please smash the like button to help our videos get discovered";
+      post.Author=editorUser;
+   }
+   ```
+
+### Configure Application Security
```

Comment:
The headings above (such as "### Declare a data model") only capitalize the first letter of the first word. Why do you capitalize "Application Security" here?

Reply:
I fixed the capitalization

Reply:
Thank you